Userland/Libraries/LibWeb/Page/EventHandler.cpp

```diff
@@ -216,6 +216,14 @@ bool EventHandler::handle_mousewheel(CSSPixelPoint viewport_position, CSSPixelPo
             auto page_offset = compute_mouse_event_page_offset(client_offset);
             if (node->dispatch_event(UIEvents::WheelEvent::create_from_platform_event(node->realm(), UIEvents::EventNames::wheel, screen_position, page_offset, client_offset, offset, wheel_delta_x, wheel_delta_y, button, buttons, modifiers).release_value_but_fixme_should_propagate_errors())) {
                 m_navigable->active_window()->scroll_by(wheel_delta_x, wheel_delta_y);
+                auto end_scroll_offset = m_navigable->active_document()->navigable()->viewport_scroll_offset();
+                auto end_position = viewport_position.translated(end_scroll_offset);
+                // After the scroll event has been dispatched,
+                // update the hovered node in order to trigger relevant events.
+                // - e.g.: mouseover, mouseenter, mouseleave, etc.
+                if (auto result = target_for_mouse_position(end_position); result.has_value()) {
+                    m_navigable->active_document()->set_hovered_node(result->paintable->dom_node());
```

Comment:
this change makes hovered node be updated as wheel events are received, while, from what I see, other browsers change hovered node after scrolling has completed. and this difference is important, because `set_hovered_node()` invalidates style, which is quite expensive. could we fix that?

Reply:
I agree it should be after scroll event has completed.

Correct me here: 

- My understanding was in 1 unit of physical scroll by user (1 single user event), `handle_mousewheel` function was called only once.
- What I'm assuming you are saying is, `handle_mousewheel` function will be called continuously as scroll is ongoing. e.g.:
   - In case of super sensitive scroll, which might last multiple seconds.
   - In 1 unit of physical scroll by user, `handle_mousewheel` will be called multiple times.